d2/src/test/java/com/linkedin/d2/xds/TestXdsClientImpl.java

```diff
@@ -311,11 +324,12 @@ public void testHandleD2URICollectionResponseWithData()
             .setResource(Any.pack(URI_BUILDER2.build()))
             .build()
     ), Collections.singletonList(URI_URN1), NONCE, null);
-    fixture._xdsClientImpl.handleResponse(DISCOVERY_RESPONSE_URI_MAP_DATA2);
-    actualData = (XdsClient.D2URIMapUpdate) fixture._clusterSubscriber.getData();
+    fixture._xdsClientImpl.handleResponse(createUri2Delete1);
+    actualData = (D2URIMapUpdate) fixture._clusterSubscriber.getData();
     // subscriber data should be updated to D2_URI_MAP_UPDATE_WITH_DATA2
-    verify(fixture._resourceWatcher).onChanged(eq(D2_URI_MAP_UPDATE_WITH_DATA2));
-    Assert.assertEquals(actualData.getURIMap(), D2_URI_MAP_UPDATE_WITH_DATA2.getURIMap());
+    D2URIMapUpdate expectedUpdate = new D2URIMapUpdate(Collections.singletonMap(URI2, URI_BUILDER2.build()));
+    verify(fixture._resourceWatcher).onChanged(eq(expectedUpdate));
+    Assert.assertEquals(actualData.getURIMap(), expectedUpdate.getURIMap());
     fixture.verifyAckSent(2);
```

Comment:
could add another case that if Uri2 is deleted next (the map becomes empty), the watcher will receive an empty map.

Reply:
Done!